vllm/worker/xpu_model_runner.py

```diff
@@ -0,0 +1,499 @@
+from typing import Dict, List, Optional, Tuple
+
+import numpy as np
+import torch
+import torch.nn as nn
+
+from vllm.attention import get_attn_backend
+from vllm.config import (DeviceConfig, LoadConfig, LoRAConfig, ModelConfig,
+                         ParallelConfig, SchedulerConfig, VisionLanguageConfig)
+from vllm.distributed import broadcast_tensor_dict
+from vllm.model_executor.model_loader import get_model
+from vllm.sampling_params import SamplingParams, SamplingType
+from vllm.sequence import SamplerOutput, SequenceData, SequenceGroupMetadata
+from vllm.utils import make_tensor_with_pad, maybe_expand_dim
+from vllm.worker.model_runner import (AttentionMetadata, SamplingMetadata,
+                                      _prepare_fake_inputs)
+
+_PAD_SLOT_ID = -1
+_BATCH_SIZE_ALIGNMENT = 8
+_BATCH_SIZES_TO_CAPTURE = [1, 2, 4] + [
+    _BATCH_SIZE_ALIGNMENT * i for i in range(1, 33)
+]
+
+
+class XPUModelRunner():
```

Comment:
It seems like the XPUModelRunner is almost the same as GPU's ModelRunner except for CUDA graph support. Can we reuse the existing ModelRunner?

Reply:
there are some gaps to support chunked prefill for xpu, so some APIs are not totally compatible yet. I will do refactor once chunked prefill get supporteded.